src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityRootAction.java

```diff
@@ -256,16 +225,22 @@ public String getUrlName() {
     public void doTestSubmitBuildRef(StaplerRequest req, StaplerResponse rsp,
             @QueryParameter(required = true) String imageId,
             @QueryParameter(required = true) String jobName) throws IOException, ServletException {
-        Run latest = Jenkins.getInstance().getItem(jobName, Jenkins.getInstance(), 
-                AbstractProject.class).getLastBuild();
+        final Jenkins j = Jenkins.getInstance();
+        if (j == null) {
+            throw new IOException("Jenkins instance is not active");
+        }
+        final AbstractProject item = j.getItem(jobName, j, AbstractProject.class);
+        final Run latest = item != null ? item.getLastBuild() : null;
+        if (latest == null) {
+            throw new IOException("Cannot find a project or run to modify"); 
+        }
+        
         DockerFingerprints.addFromFacet(null,imageId, latest);
```

Comment:
until this is removed it should at least check the user has build permissions on this project.